compiler/rustc_codegen_llvm/src/base.rs

```diff
@@ -114,14 +123,6 @@ pub fn compile_codegen_unit(tcx: TyCtxt<'_>, cgu_name: Symbol) -> (ModuleCodegen
                 cx.coverageinfo_finalize();
             }
 
-            // Create the llvm.used and llvm.compiler.used variables.
-            if !cx.used_statics().borrow().is_empty() {
-                cx.create_used_variable()
-            }
-            if !cx.compiler_used_statics().borrow().is_empty() {
-                cx.create_compiler_used_variable()
-            }
-
```

Comment:
Reordering this seems suspect given comment above:

>Finalize code coverage by injecting the coverage map. Note, the coverage map will also be added to the `llvm.compiler.used` variable, created next.

Reply:
Oh dear, nice catch.